Sustainable society

alyamahmoud opened the pull request:
I have created: 
- template notebook: students should fill in with their own analysis
- example notebook: uses music dataset to propose and answer some questions

I would appreciate your feedback on both, particularly: 

- libraries used
- level of complexity

thank you

lisancao committed:
initial commit

lisancao committed:
refactor, add data, add code

lisancao committed:
add pets and tutorial files

lisancao committed:
file refactor, add notebook

lisancao committed:
add question 1A

lisancao committed:
add data in directory, complete challenge 1

lisancao committed:
moved challenge interactives to py script, minor grammar fixes and add fill in code

lisancao committed:
begin challenge 2, add files and functions

alyamahmoud committed:
hackathon templates

alyamahmoud committed:
Merge branch 'master' of https://github.com/callysto/hackathon

alyamahmoud committed:
hackathon templates

alyamahmoud committed:
hackathon example 01

misterhay commented:
I think they look good. We've been trying to following the callysto/curriculum-notebooks naming conventions of CamelCase directory names and kebab-case for files, but that's probably not a big deal.

In **Section II** of the music example we can probably leave out the phrase "available as a data frame/table" to simplify the wording. And there's an extra `\` in the third paragraph.

And the csv link should probably be `https://raw.githubusercontent.com/callysto/hackathon/sustainable-society/SustainabilityOnMars/spotify_data_2000_2020.csv` or perhaps `./spotify_data_2000_2020.csv`. We may also want to move `music.head()` to the same cell as `pd.read_csv` so that the cell displays output when it runs.

Instead of using numpy in `np.max(music.duration_ms)` it might be more intuitive to use the pandas method `music['duration_ms'].max()`.

Before the first chart (Audio characteristics over years) it might be good to have a markdown cell describing what the visualization will be and why we are creating it.

"What's next !" should probably be "What's next?" or perhaps "Next Steps"

These are all minor issues or suggestions, though, the notebooks look good.

lisancao commented:
In the conclusion cell "shorter" -> shorter 

Having comments for each line in the plot code would be helpful for beginners. We also had a bit of an issue with the conclusions in the notebooks being too short or too long, maybe adding more to it and adding what makes a good conclusion, such as why we want to look at duration in the first place. 

For the template, maybe have the examples on the same line as the question to make it more compact? then add an empty markdown cell below that so they can put their question in there. The heading size for Observations section is #### instead of ###. Otherwise looks good!

lisancao committed:
add text colouring to Q1 and embed youtube videos

lisancao committed:
add Challenge2 code and youtube videos

lisancao committed:
more content, sectioned py script better

lisancao committed:
add tutorial, more content

- add place answer below line for cells
- add challenge 2b question
- added copy of pandas cheatsheet to directory

lisancao committed:
directory rename to camelcase

lisancao committed:
code fix and temp cell added

lisancao committed:
add outline for rest of notebook

lisancao committed:
cleared output

lisancao committed:
add notebook-basics

lisancao committed:
add challenge2c, content, minor fixes

lisancao committed:
small content additions

lisancao committed:
add content

alyamahmoud committed:
Rename hackathon_example_01_music.ipynb to music-example.ipynb

alyamahmoud committed:
CamelCase for directories and kebab-case for files

alyamahmoud committed:
Create 1

alyamahmoud committed:
Add files via upload

CamelCase for directories and kebab-case for files, directory structure

alyamahmoud committed:
Delete 1

alyamahmoud committed:
Create 1

alyamahmoud committed:
Add files via upload

CamelCase for directories and kebab-case for files, directory structure

alyamahmoud committed:
Delete 1

alyamahmoud committed:
update example notebook

alyamahmoud committed:
update template notebook

alyamahmoud commented:
updated template and example notebooks accordingly.
